src/main/java/com/newestuser/imen/bgdate/BulgarianDate.java

```diff
@@ -0,0 +1,174 @@
+
+    /**
+     * Converts the passed gregorian day of the month to the day according to the Bulgarian calendar.
+     *
+     * @param currentDay current day in a given gregorian month.
+     * @param bgMonth    current month according to the Bulgarian calendar.
+     * @param month      current month according to the Gregorian calendar.
+     * @return the converted Bulgarian day.
+     */
+    private static int findBulgarianDay(int currentDay, String bgMonth, int month) {
+        DayMonthRange dayMonthRange = GREGORIAN_MONTH_TO_BG_MONTH.get(month);
+        bgMonth = dayMonthRange.getBulgarianMonth(currentDay);
+        int bulgarianDateDay = currentDay;
+        if (DAY_CONVERSION_VALUES.containsKey(bgMonth)) {
+            if (dayMonthRange.findIsInFirstMonthPart(currentDay)) {
+                bulgarianDateDay += DAY_CONVERSION_VALUES.get(bgMonth)[0];
+            } else {
+                bulgarianDateDay -= DAY_CONVERSION_VALUES.get(bgMonth)[1];
+            }
+        }
+        if ((month > 2) && !findIsLeapYear(Year.now().getValue())) {
+            bulgarianDateDay -= 1;
+        }
+        return bulgarianDateDay;
+    }
+
+    /**
+     * Converts the passed gregorian month to the month according to the Bulgarian calendar.
+     *
+     * @param month      current month according to the Gregorian calendar.
+     * @param currentDay current day in a given gregorian month.
+     * @return the converted Bulgarian month.
+     */
+
+    private static String convertMonth(int month, int currentDay) {
+        if (month == 12 && currentDay == 21) {
+            return "Нова година";
+        }
+        DayMonthRange dayMonthRange = GREGORIAN_MONTH_TO_BG_MONTH.get(month);
+        return dayMonthRange.getBulgarianMonth(currentDay);
+    }
+
+
+    /**
+     * Determines the passed Bulgarian year into Bulgarian year type.
+     *
+     * @param bulgarianDateYear current year according to the Bulgarian calendar.
+     * @return Bulgarian type of year.
+     */
+
+    private static String findYearType(int bulgarianDateYear) {
+        return YEAR_TYPES.get(bulgarianDateYear % 12);
+    }
+
+    /**
+     * Determines the passed Gregorian year is leap.
+     *
+     * @param currentYear current year according to the Gregorian calendar.
+     * @return true if the passed year is leap.
+     */
+
+    private static boolean findIsLeapYear(int currentYear) {
+        return (currentYear % 4 == 0 && currentYear % 100 != 0) || (currentYear % 400 == 0);
+    }
+
+    /**
+     * Creates current Bulgarian date.
+     * Bulgarian year is calculated as follows: current year + 4768.
+     *
+     * @return BulgarianDate object with the current date.
+     */
+
+    public static BulgarianDate now() {
+        BulgarianDate bulgarianDate = new BulgarianDate();
+        bulgarianDate.bulgarianDateDay = findBulgarianDay(Calendar.getInstance().get(Calendar.DAY_OF_MONTH), bulgarianDate.bulgarianDateMonth, Calendar.getInstance().get(Calendar.MONTH) + 1);
```

Comment:
I am not sure why you pass `bulgarianDtae.buglarianDateMonth` as a parameter to `findBulgarianDay` when you attempt to calculate the bulgarianDay when that parameter is not even initialized.

You can remove that parameter from `findBulgarianDay` you do not use it.

Reply:
Overall all this entire logic is duplicated in the constructor.

If you chain the constructors this method can become:

```java
public static BulgarianDate now() {
   return new BulgarianDate();
}
```

And you would be able to remove the `static` modified from all the private methods.